tf_quant_finance/black_scholes/vanilla_prices_test.py

```diff
@@ -134,6 +134,34 @@ def test_price_vol_and_expiry_scaling(self):
             forwards=forwards))
     self.assertArrayNear(base_prices, scaled_prices, 1e-10)
 
+  def test_option_prices_detailed_discount(self):
+    """Tests that the prices are when giving risk_free_rates and
+    cost_of_carries."""
+    spots = np.array([80.0, 90.0, 100.0, 110.0, 120.0] * 2)
+    strikes = np.array([100.0] * 10)
+    risk_free_rates = 0.08
+    volatilities = 0.2
+    expiries = 0.25
+
+    is_call_options = np.array([True] * 5 + [False] * 5)
+    cost_of_carries = -0.04
+    computed_prices = self.evaluate(
+      tff.black_scholes.option_price(
+        volatilities,
+        strikes,
+        expiries,
+        spots=spots,
+        risk_free_rates=risk_free_rates,
+        cost_of_carries=cost_of_carries,
+        is_call_options=is_call_options,
+        dtype=tf.float32))
+    fake_expected_prices = np.array([0.028111, 0.561231, 3.401953, 9.831990,
```

Comment:
why `fake_expected_prices`?

Reply:
fake_expected_prices are the actual current output of the functions, while expected_prices is what is recorded in the paper. Would be great if you could just look through them and make sure you're happy with the discrepancies. Then we can keep either one as the 'truth' for testing in the future.

Reply:
Fixed in your other pull request